lib/rom/sql/gateway.rb

```diff
@@ -177,11 +181,18 @@ def connect(uri, *args)
         end
       end
 
-      # Add extensions to the database connection
+      # Load database-specific extensions
       #
       # @api private
-      def add_extensions(exts)
-        connection.extension(*exts)
+      def load_extensions(exts)
+        db_type = connection.database_type.to_sym
+
+        if ROM::SQL.available_extension?(db_type)
+          ROM::SQL.load_extensions(db_type)
+        end
+
+        extensions = (CONNECTION_EXTENSIONS.fetch(db_type) { [] } + exts).uniq
```

Comment:
That's a good question. I did it automatically tbh, but the difference is that `, [])` will always allocate create a new array and `{ [] }` won't. Should be `fetch(db_type, EMPTY_ARRAY)` instead of both variants :)